presto-main/src/main/java/com/facebook/presto/sql/planner/plan/AggregationNode.java

```diff
@@ -359,16 +358,28 @@ public static Step partialInput(Step step)
     {
         private final FunctionCall call;
         private final FunctionHandle functionHandle;
+        private final List<Expression> arguments;
+        private final Optional<Expression> filter;
+        private final Optional<OrderingScheme> orderingScheme;
+        private final boolean isDistinct;
         private final Optional<Symbol> mask;
 
         @JsonCreator
         public Aggregation(
                 @JsonProperty("call") FunctionCall call,
                 @JsonProperty("functionHandle") FunctionHandle functionHandle,
+                @JsonProperty("arguments") List<Expression> arguments,
```

Comment:
I don't think you need to move out `arguments`... Otherwise `FunctionCall` is nothing but a name. Keep `arguments` inside `call` can help the refactoring of directly replacing `FunctionCall` with `CallExpression`

Reply:
Yes, it was temporarily moved out to be able to convert it to RowExpression (just to make it easier to code). In the end argument is part of CallExpression. Once these squashed, it should not be a problem.